Anlab.Mvc/Client/components/AdditionalEmails.tsx

```diff
@@ -52,30 +61,59 @@ export class AdditionalEmails extends React.Component<IAdditionalEmailsProps, IA
 
         const emails = this.props.addedEmails.map(item => {
             return (
-                <div key={item}>{item} <Button label="Delete" primary onClick={() => this.onDelete(item)}/></div>
+                <Chip key={item} deletable onDeleteClick={() => this.onDelete(item)} > { item }</Chip>
             );
         });
 
 
         return emails;
     }
 
-    _renderAddButton = () => {
-        if (this.state.email.length > 0) {
-            return (<Button label='Add Email' flat primary onClick={this.onClick} />);
-        } else {
-            return null;
+    handleKeyPress = (e) => {
+        if (e.key === 'Enter') {
+            if (this.state.email === "")
+                this._toggleAddEmail();
+            else
+                this.onClick();
+        }
+    }
+
+    handleBlur = () => {
+        if (this.state.email != "") {
+            this.onClick();

```

Comment:
@laholstege nit: extract the onClick method to a new method, maybe `_addEmailAddress` or similar, then call from other methods.  That way you don't have to call this.onClick() even for things like keypress and blur that isn't really clicks